redfish_client/caching_connector.py

```diff
@@ -0,0 +1,37 @@
+#  Copyright 2019 XLAB d.o.o.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from redfish_client.connector import Connector
+
+
+class CachingConnector(Connector):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._cache = {}
+
+    def get(self, path):
```

Comment:
No need to do it right now, but would it make sense to introduce optional argument `bypass_cache=False` so that functions like `wait_for_stats` don't need to reset entire cache?

Reply:
EDIT: ah, I see we can achieve similar with `reset(/path)`. Works as well

Reply:
Functions such as `wait_for_stats` should use higher-level API such as `Resource.refresh()` that handles all the details.